### DISCUSSION TITLE
06 transfer learning using torch.compile()

### DISCUSSION THREAD

**Discussion by sacromentus** [mrdbourke/pytorch-deep-learning#682] [General]
I tried implementing torch.compile() on the EfficientNet model, i did model = torch.compile(model) right after I instantiate model, but it does not work. First, torchsummary doesn't work with it, which I read is normal. But when I train the model, it won't train and I get various errors, one of them being "BackendCompilerFailed: backend='inductor' raised:"

Has anyone implemented it successfully?

**Comment by mrdbourke** [comment]
Hi @sacromentus , 

I haven't tried too much transfer learning with `torch.compile()` just yet.

But I would like to.

Have you tried a different backend with `torch.compile()`? See: https://pytorch.org/docs/stable/generated/torch.compile.html 

Also, which hardware are you running on? That *might* effect how `torch.compile()` performs?

Have you managed to get the model to do a single forward pass? That might help with troubleshooting.

**Reply by sacromentus** [reply]
Daniel,

Thanks for the reply and the amazing course! I will try a different backend; also, I will try it on some of the models we built instead of a pre-trained one. I’ll report back my results.My hardware is a laptop NVIDIA 4070 (cuda is enabled, and boy was that a learning experience in and of itself!)

-Jerry

**Reply by mrdbourke** [reply]
Hahaha you're so right! Setting up CUDA for the first time can be a bit of a challenge.

Glad to hear you got it working :) 

Also I'd be keen to hear if you get `torch.compile()` working with a pre-trained model too!
